static/js/reactions.js

```diff
@@ -0,0 +1,194 @@
+    if (user_list.length === 1) {
+        event.count = 1;
+        event.title = full_name(event.user.user_id);
+        $(selector).append(templates.render('message_reaction', event));
+        reaction.prop('title', full_name(event.user.user_id));
+    } else {
+        var count_element = reaction.find('.message_reaction_count');
+        var new_title = reaction.attr('title') + ', ' + full_name(event.user.user_id);
+        reaction.prop('title', new_title);
+        count_element.html(user_list.length);
+    }
+}
+
+function remove_reaction(event) {
+    var emoji_name = event.emoji_name;
+    var message_id = event.message_id;
+    var stored_reactions = reactions_store.get(message_id);
+    if (!stored_reactions.has(emoji_name)) {
+      return;
+    }
+    var user_list = stored_reactions.get(emoji_name);
+    var i = user_list.indexOf(event.user.user_id);
+    if (i !== -1) {
+        user_list.splice(i);
+    }
+    var new_title = get_title_from_user_ids(user_list);
+    var selector = '#zhome' + message_id + ' .message_reactions';
+    var matching_reactions = $(selector).find('[data-emoji-name="' + emoji_name + '"]');
+    var count_element = matching_reactions.find('.message_reaction_count');
+    matching_reactions.title = new_title;
+    count_element.html(user_list.length);
+    if (user_list.length === 0) {
+        matching_reactions.remove();
+    }
+}
+
+function get_emojis_used_by_user_for_message_id(message_id) {
+    var user_id = page_params.user_id;
+    var stored_reactions = reactions_store.get(message_id);
+    if (!stored_reactions) {
+        return [];
+    }
+    var ret = stored_reactions.keys().filter(function (emoji_name) {
+        var i = stored_reactions.get(emoji_name).indexOf(user_id);
+        return i !== -1;
+    });
+    return ret;
+}
+
+function update_reaction_store(messages) {
+    _.each(messages, function (message) {
+        var stored_reactions = reactions_store.setdefault(message.id, new Dict());
+        _.each(message.reactions, function (reaction) {
+            var user_list = stored_reactions.setdefault(reaction.emoji_name, []);
+            user_list.push(reaction.user.id);
+        });
+    });
+}
+
+function get_reactions_for_message_id(message_id) {
+    return reactions_store.get(message_id).items().map(function (item) {
+        return {
+            emoji_name: item[0],
+            count: item[1].length,
+            title: get_title_from_user_ids(item[1])
+        };
+    });
+}
+
+function get_reactions_for_message(message) {
+    update_reaction_store([message]);
+    return get_reactions_for_message_id(message.id);
+}
+
+exports = {
+    add_reaction: add_reaction,
+    remove_reaction: remove_reaction,
+    get_emojis_used_by_user_for_message_id: get_emojis_used_by_user_for_message_id,
+    get_reactions_for_message: get_reactions_for_message,
+    update_reaction_store: update_reaction_store,
+    message_reaction_on_click: message_reaction_on_click
+};
```

Comment:
Can you redo the exports in the same style as our other JavaScript modules, where what is exported is declared in the function declaration?